### PR TITLE
Feat/332 replace orchard input with a combobox

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Completes #332

#### Main changes:
Integrate front-end with the new get orchards by vegCode endpoint, and display the options to users in comboboxes, which replaced the old fetch orchard by id flow. 

#### Other changes:
- rename `OrchardModal` to `OrchardWarnModal` and other variables/functions
- stop showing warn modal when adding an orchard
- stop displaying the add/delete buttons in form review for orchard step
- refine the logic for determining the fetch status of `parent-tree-genetic-quality` queries 
- add default behaviour to React Query so we don't have to specify each query to not fetch on window refocus and remount. 
    - ```json
        "queries":  {
            "refetchOnMount": false,
            "refetchOnWindowFocus": false,
            "retry": 3
        }
        ``` 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
local test


## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged


---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-352-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-352-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-352-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-352-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-352-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-352-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/index.tsx [hunk 1]
```diff
@@ -20,7 +20,17 @@ if (isDevEnv) {
...
 
-const queryClient = new QueryClient();
+const queryClient = new QueryClient(
+  {
+    defaultOptions: {
+      queries: {
+        refetchOnMount: false,
+        refetchOnWindowFocus: false,
+        retry: 3
+      }
```
Comment (mgaseta) [inline frontend/src/index.tsx:30]: Very classy! :tophat: :monocle_face:

File: frontend/src/components/SeedlotRegistrationSteps/OrchardStep/definitions.ts [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
+  controlledCross: boolean,
+  biotechProcess: boolean,
+  noPollenContamination: boolean,
+  breedingPercentage: string,
+  pollenMethodology: boolean
+}
+
+export type StagedOrchardOpt = {
+  inputId: number,
+  selectedItem: MultiOptionsObj | null
```
[the remark this answers is not in the document]
Comment (craigyu) [reply frontend/src/components/SeedlotRegistrationSteps/OrchardStep/definitions.ts]: lol good catch, it's removed [here](https://github.com/bcgov/nr-spar/pull/352/commits/13a069d2ca491385843285236e0e9b8464a9353f)

**Review by mgaseta — APPROVED** [review]
Nice job! :1st_place_medal: :+1: 
I think we can merge this change and open a new ticket for that strange bug, it seems to happen only in very specific cases so it can wait a while.

**Comment by craigyu** [comment]
@mgaseta has approved the PR so I'm bypassing the requirement of needing a code owner's approval

**Pull request merged**